cookbook/integrating-custom-filters.rst

```diff
@@ -0,0 +1,117 @@
+defines the url where the fields api can be found. The class parameter defines 
+the associated entity.In the features the filter has to be enabled.
+
+.. code-block:: yaml
+
+	sulu_resource:
+	    contexts:
+	        contact:
+	            fields: "/admin/api/contacts/fields"
+	            class: "Sulu\Bundle\ContactBundle\Entity\Contact"
+	            features:
+	                - "filters"
+
+3. Extend the js configuration in your bundle
+
+Extend the js configuration in your bundle with a config value for the bread-
+crumb and an url for the fields api. The last part of the setting key is the
+context the filters component will use.
+
+.. code-block:: js
+
+    Config.set('suluresource.filters.type.contacts', {
+        breadCrumb: [
+            {title: 'navigation.contacts'},
+            {title: 'contact.contacts.title'}
+        ],
+        fields: 'admin/api/contacts/fields'
+    });
+
+5. Extend the list and toolbar initialization for the lists
+
+The toolbar should have at least two groups and one of them should have the id 
+2 because the filter button will be added in the one with id 2. Two additional
+parameters have to added at the end. The first one is the instance name of the
+datagrid and the second one is the selector for the container where the result
+of the filter (x entries match filter y) will be displayed. Therefore a div
+above the filter div should be added in the html.
+
+.. code-block:: js
+
+	this.sandbox.sulu.initListToolbarAndList.call(this, 'contacts', '/admin/api/contacts/fields',
+        {
+            el: this.$find('#list-toolbar-container'),
+            instanceName: 'contacts',
+            inHeader: true,
+            groups: [
+                {
+                    id: 1,
+                    align: 'left'
+                },
+                {
+                    id: 2,
+                    align: 'right'
+                }
+            ]
+        },
+        {
+            el: this.sandbox.dom.find('#people-list', this.$el),
+            url: '/admin/api/contacts?flat=true',
+            searchInstanceName: 'contacts',
+            searchFields: ['fullName'],
+            resultKey: 'contacts',
+            instanceName: 'contacts',
+            viewOptions: {
+                table: {
+                    icons: [
+                        {
+                            icon: 'pencil',
+                            column: 'firstName',
+                            align: 'left',
+                            callback: function(id) {
+                                this.sandbox.emit('sulu.contacts.contacts.load', id);
+                            }.bind(this)
+                        }
+                    ],
+                    highlightSelected: true,
+                    fullWidth: true
+                }
+            }
+        },
+        'contacts',
+        '#people-list-info'
+    );
+
+.. code-block:: html
+
+	{% extends "SuluContactBundle:Template:base.list.html.twig" %}
```

Comment:
@michaelzangerle please add here a "normal" example without inheritance of templates